src/service/resources-impl.js

```diff
@@ -1741,28 +1816,57 @@ export class Resource {
   }
 
   /**
-   * Calls element's documentInactiveCallback callback and resets state for
+   * Calls element's unlayoutCallback callback and resets state for
    * relayout in case document becomes active again.
    */
-  documentBecameInactive() {
-    if (this.state_ == ResourceState_.NOT_BUILT) {
+  unlayout() {
+    if (this.state_ != ResourceState_.LAYOUT_COMPLETE) {
       return;
     }
     if (this.isInViewport()) {
       this.setInViewport(false);
     }
-    if (this.element.documentInactiveCallback()) {
+    if (this.element.unlayoutCallback()) {
       this.state_ = ResourceState_.NOT_LAID_OUT;
       this.layoutCount_ = 0;
     }
   }
 
   /**
-   * Called when a previously visible element has become invisible.
+   * Calls element's pauseCallback callback.
+   */
+  pause() {
+    if (this.state_ == ResourceState_.NOT_BUILT) {
+      return;
+    }
+    if (this.isInViewport()) {
+      this.setInViewport(false);
+    }
+    this.element.pauseCallback();
```

Comment:
I'm wondering if it'd be helpful to also set state to `this.state_ = ResourceState_.PAUSED`. It's in theory a correct state. But it depends on whether we need to condition resume action on this state, as I asked in one of the comments above.